### PR TITLE
HSEARCH-3484 + HSEARCH-3485 + HSEARCH-3491: JDK12 compatibility on branch 5.11

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3484
https://hibernate.atlassian.net//browse/HSEARCH-3485
https://hibernate.atlassian.net//browse/HSEARCH-3491
https://hibernate.atlassian.net//browse/HSEARCH-3494

Full build (including JDK12): http://ci.hibernate.org/job/hibernate-search/job/PR-1894/4/
Note that OSGi and WildFly integration tests are disabled on JDK12, because Karaf and WildFly don't work on JDK12 yet.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -387,6 +387,9 @@
         <surefire.jvm.args>${surefire.jvm.args.memory} ${surefire.jvm.args.java-version} ${surefire.jvm.args.jacoco}</surefire.jvm.args>
         <failsafe.jvm.args>${surefire.jvm.args.memory} ${surefire.jvm.args.java-version} ${failsafe.jvm.args.jacoco}</failsafe.jvm.args>
 
+        <failsafe.wildfly.skip>false</failsafe.wildfly.skip>
+        <failsafe.osgi.skip>false</failsafe.osgi.skip>
```
Comment (gsmet) [inline pom.xml]: It seems you forgot to set them to true for the JDK 12+ profile? Or did I miss something?

Comment (yrodiere) [reply pom.xml]: I did that in the Jenkinsfile. This way, if someone tries to build by hand, they will see that it doesn't work (yet) on WF and OSGi, but our CI tests will still check everything else.

Comment (gsmet) [reply pom.xml]: Hum. Not sure it's the desired behavior. That means people won't be able to build our code with the latest JDK.

Comment (yrodiere) [reply pom.xml]: That was not my goal at all... but ok, I'll do it your way.

**Comment by yrodiere** [comment]
Merged. Thanks!

**Pull request merged**